app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -279,28 +216,51 @@ class LuciBuildService {
     final List<Request> requests = <Request>[];
     final List<String> branches =
         await gerritService.branches('flutter-review.googlesource.com', 'recipes', 'flutter-');
+
+    final String sha = pullRequest.head!.sha!;
+    String cipdVersion = 'refs/heads/${pullRequest.base!.ref!}';
+    cipdVersion = branches.contains(cipdVersion) ? cipdVersion : config.defaultRecipeBundleRef;
+
     for (Target target in targets) {
-      final BuilderId builderId = BuilderId(
-        project: 'flutter',
-        bucket: 'try',
-        builder: target.value.name,
+      final github.CheckRun checkRun = await githubChecksUtil.createCheckRun(
+        config,
+        target.slug,
+        sha,
+        target.value.name,
       );
-      final Map<String, dynamic> userData = <String, dynamic>{
-        'repo_owner': pullRequest.base!.repo!.owner!.login,
-        'repo_name': pullRequest.base!.repo!.name,
-        'user_agent': 'flutter-cocoon',
+
+      final github.RepositorySlug slug = pullRequest.base!.repo!.slug();
+
+      final Map<String, dynamic> userData = <String, dynamic>{};
+      if (checkSuiteEvent != null || config.githubPresubmitSupportedRepo(slug)) {
+        userData['check_run_id'] = checkRun.id;
+        userData['commit_sha'] = sha;
+        userData['commit_branch'] = pullRequest.base!.ref!.replaceAll('refs/heads/', '');
+        userData['builder_name'] = target.value.name;
+      }
+
+      Map<String, List<String>> tags = <String, List<String>>{
+        'github_checkrun': <String>[checkRun.id.toString()],
       };
-      requests.add(await _createBuildRequest(
-        checkSuiteEvent: checkSuiteEvent,
-        pullRequest: pullRequest,
-        builder: target.value.name,
-        builderId: builderId,
-        userData: userData,
-        properties: target.getProperties(),
-        dimensions: target.getDimensions(),
-        branches: branches,
-      ));
+
+      Map<String, Object> properties = target.getProperties();
+      properties.putIfAbsent('git_branch', () => pullRequest.base!.ref!.replaceAll('refs/heads/', ''));
+
+      // Are builder and checkrun the same?
```

Comment:
No. CheckRuns contain only their name, id, and commit info. Builders only contain the name.

An easier way to think about it is a CheckRun can happen multiple times, whereas the Builder is a group of check runs with the same name. Similarly for post-submit, it's how we have Tasks and Targets

Reply:
This is really good information. Thank you for that!